src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_ev_driver_posix.cc

```diff
@@ -41,14 +41,15 @@ namespace grpc_core {
 class GrpcPolledFdPosix : public GrpcPolledFd {
  public:
   GrpcPolledFdPosix(ares_socket_t as, grpc_pollset_set* driver_pollset_set)
-      : as_(as) {
+      : as_(as), driver_pollset_set_(driver_pollset_set) {
     gpr_asprintf(&name_, "c-ares fd: %d", (int)as);
     fd_ = grpc_fd_create((int)as, name_, false);
     grpc_pollset_set_add_fd(driver_pollset_set, fd_);
   }
 
   ~GrpcPolledFdPosix() {
     gpr_free(name_);
+    grpc_pollset_set_del_fd(driver_pollset_set_, fd_);
```

Comment:
Do we have some guarantee that this pollset_set will still exist when this fd is destroyed?

Reply:
Good point, I don't think so. Canged the resolver to create and destroy it's own pollset set.

Reply:
Yeah I agree, and my reason for thinking this is the right approach is by looking at ownership of the fd's: the `AresDnsResolver`'s pollset set is destroyed upon `AresDnsResolver` [destruction](https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc#L166). The entire c-ares query, which creates the event driver and fd's, is protected by a [ref](https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc#L411) that is released upon the query's completion. That completion begins [its scheduling](https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc#L159) from a callback that's invoked during a call to `ares_process` on an fd though, and then remaining fd's in the driver's list are scheduled to be shut-down/cancelled/destroyed. So it would appear that the original `AresDnsResolver` might have been unref'd by the time that a <i>second</i> driver fd is destroyed. It seems tricky to test for this though...